src/mpid/ch4/shm/ipc/src/ipc_send.h

```diff
@@ -24,17 +23,18 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_IPCI_try_lmt_isend(const void *buf, MPI_Aint
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_IPCI_TRY_LMT_ISEND);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_IPCI_TRY_LMT_ISEND);
 
+    /* NOTE: currently the VCI(0).lock is protecting the entire shm path. The ipc_attr may
+     * need its own granular locks once we add multi-vci support.
+     */
     MPID_THREAD_CS_ENTER(VCI, MPIDI_VCI(0).lock);
```

Comment:
I don't understand the new comment. Why do we need separate lock for `ipc_attr`? It is only a local variable, read-only to SHM code.

Reply:
Let me just delete the comment. I don't think the comment adds more clarity. I think @KaimingOuyang 's intent is to comment that the routine to look up `ipc_attr` may need different granular locks than the per-vci lock. It is not clear how we gonna do multiple-vci on shm-path yet so any more specific comment may turn out to be misleading.

Reply:
I just want to add a reminder here. If in future, VCI CS allows multiple threads to call `MPIDI_GPU_get_ipc_attr`, that will cause a problem since `MPIDI_GPU_get_ipc_attr` is not thread-safe. If we current VCI design is still ongoing, let's remove it to reduce the confusion.

Reply:
Let us do not add comment that depends on uncertain future code. The description is unclear and causes confusion.